examples/misc/generics/list.bi

```diff
@@ -0,0 +1,66 @@
+#macro list(T)
+	__fb_join__(list_,__fb_arg_rightof__(T,OF))
+#endmacro
+
+#macro list_of(T)
+	list_##T
+#endmacro
+
+#macro imp_list_of(T)
+#if not defined(list_##T)	
+	type list_item_##T
+		value as T
+		next_ as list_item_##T ptr
+	end type
+
+	type list_##T
+		declare constructor(initial as integer = 0)
+		declare destructor()
+		declare sub add(v as T)
+		declare operator[](idx as integer) as T

```

Comment:
This might be a candidate for `byref` result (`declare operator[](idx as integer) byref as T`)